rust/compaction/src/details.rs

```diff
@@ -176,6 +176,27 @@ pub async fn merge_sorted_files(input_data: &CompactionInput<'_>) -> Result<Comp
     }
 }
 
+async fn create_object_store_factory(
+    aws_config_override: &Option<AwsConfig>,
+) -> ObjectStoreFactory {
+    let s3_config = match aws_config_override {
```

Comment:
These double nested match blocks can be re-written for ease of reading. For the inner one, consider using Result::ok() to convert from a Result to an Option: https://doc.rust-lang.org/std/result/enum.Result.html#method.ok E.g. default_s3_config().await.ok()

The outer match should use Option::map to convert a value (if present). Map::or and Map::or_else are useful for converting Option::None to something else.

Reply:
I've done that, thanks very much! It's a bit awkward that Option::or doesn't work with async/await.